h/views/activity.py

```diff
@@ -296,17 +296,26 @@ def delete_lozenge(self):
     def toggle_tag_facet(self):
         return _toggle_tag_facet(self.request)
 
+    def _user_can_join(self):
+        """
+        Decide whether to show a page which will prompt the user to join.
+
+        :returns True if the user has permission to join or they are not logged in
+        and False otherwise.
+        """
+        if not self.request.user:
+            return True
+        return self.request.has_permission('join', self.group)
```

Comment:
I think just `"""Return True if the user has permission to join the group, False otherwise."""` would be a better docstring for this. It's not actually this function that decides whether to show a join page, this function just returns `True` or `False`, the code below that calls this function is deciding whether to show a join page.